app/queries/find_exact_match.rb

```diff
@@ -0,0 +1,39 @@
+class FindExactMatch
+  def initialize(query_field: :tags)
+    raise ArgumentError.new("query_field must be `:tags` or `:links`") unless %i{tags links}.include?(query_field)
+
+    @query_field = query_field.to_sym
+  end
+
+
+  def call(query_hash, document_type)
```

Comment:
I understand your reasoning here, however I would prefer to leave this as is for the moment as we will need to add `email_document_supertype` and `government_document_supertype` to the argument list. I would prefer to update the interface when we do that PR. (which is what I am working on today)